setup/setup.py

```diff
@@ -259,60 +206,133 @@ def createIndex():
     print("Account created!")
 
-try:
-    import logging
-    # elasticsearch logs lots of warnings on retries/connection failure
-    logging.getLogger("elasticsearch").setLevel(logging.ERROR)
-    createIndex()
-    
-     
-except Exception as e:
-    print("Index creation failed: %s" % e)
-    sys.exit(1)
 
-kibble_yaml = '../api/yaml/kibble.yaml'
+def get_kibble_yaml() -> str:
+    """Resolve path to kibble config yaml"""
+    kibble_yaml = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        os.pardir,
+        "api",
+        "yaml",
+        "kibble.yaml"
+    )
+    if os.path.exists(kibble_yaml):
+        print(f"{kibble_yaml} already exists! Writing to {kibble_yaml}.tmp instead")
+        kibble_yaml = kibble_yaml + ".tmp"
+    return kibble_yaml
 
-if os.path.exists(kibble_yaml):
-    print("%s already exists! Writing to %s.tmp instead" % (kibble_yaml, kibble_yaml))
-    kibble_yaml = kibble_yaml + ".tmp"
-    
 
-print("Writing Kibble config (%s)" % kibble_yaml)
+def save_config(
+    mlserver: str,
+    hostname: str,
+    port: int,
+    dbname: str,
+):
+    """Save kibble config to yaml file"""
+    if ":" in mlserver:
+        try:
+            mailhost, mailport = mlserver.split(":")
+        except ValueError:
+            raise ValueError("mailhost argument must be in form of `host:port` or `host`")
+    else:
+        mailhost = mlserver
+        mailport = 25
 
-m = mlserver.split(':')
-if len(m) == 1:
-    m.append(25)
-    
-myconfig = {
-    'api': {
-        'version': KIBBLE_VERSION,
-        'database': KIBBLE_DB_VERSION
-    },
-    'elasticsearch': {
-        'host': hostname,
-        'port': port,
-        'ssl': False,
-        'dbname': dbname
-    },
-    'mail': {
-        'mailhost': m[0],
-        'mailport': m[1],
-        'sender': 'Kibble <noreply@kibble.kibble>'
-    },
-    'accounts': {
-        'allowSignup': True,
-        'verify': True
+    config = {
+        'api': {
+            'version': KIBBLE_VERSION,
+            'database': KIBBLE_DB_VERSION
+        },
+        'elasticsearch': {
+            'host': hostname,
+            'port': port,
+            'ssl': False,
+            'dbname': dbname
+        },
+        'mail': {
+            'mailhost': mailhost,
+            'mailport': int(mailport),
+            'sender': 'Kibble <noreply@kibble.kibble>'
+        },
+        'accounts': {
+            'allowSignup': True,
+            'verify': True
+        }
     }
-}
 
-with open(kibble_yaml, "w") as f:
-    f.write(yaml.dump(myconfig, default_flow_style = False))
-    f.close()
+    kibble_yaml = get_kibble_yaml()
+    print(f"Writing Kibble config to {kibble_yaml}")
+    with open(kibble_yaml, "w") as f:
+        f.write(yaml.dump(config, default_flow_style = False))
+        f.close()
+
+
+def get_user_input(msg: str):
+    value = None
+    while not value:
+        value = input(msg)
```

Comment:
Here the password will be echoed during typing. 
What do think about `getpass()`?
https://docs.python.org/3.5/library/getpass.html

But also email address will be hidden with `gepass()`.

```suggestion
        value = input(msg)
```

Reply:
Good one, applied